### PR TITLE
DrivAerML: T_max=36 + lr=4.8e-4 + gc=0.3 MSE-only (softer clipping compound)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

Softer gradient clipping (gc=0.3 vs gc=0.5) at the lr=4.8e-4 + T_max=36 compound may find a deeper basin. At higher LR, the cosine schedule peaks generate larger gradient spikes. gc=0.3 may stabilize these without over-constraining optimization.

## Instructions

Run DrivAerML with the lr=4.8e-4+T_max=36 compound config with **gc=0.3** instead of gc=0.5:

```bash
cd target/icml2026 && SENPAI_TIMEOUT_MINUTES=600 python train.py \
  --dataset drivaerml \
  --optimizer adamw \
  --lr 4.8e-4 \
  --cosine-t-max 36 \
  --grad-clip 0.3 \
  --enable-fourier \
  --model-layers 4 --model-hidden-dim 512 --model-heads 8 \
  --epochs 999 --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 --max-eval-batches 200 \
  --ema-decay 0.9995 --ema-mode fixed \
  --no-compile-model --save-checkpoint \
  --wandb-group dm-lr48e4-tmax36 \
  --agent franky \
  --wandb-name "franky/dm-lr48e4-tmax36-gc03"
```

Key changes: `--lr 4.8e-4 --cosine-t-max 36 --grad-clip 0.3` (MSE-only, no --dm-loss flag)

If a checkpoint is saved, run full-eval for paper-facing TEST metric: add `--max-eval-batches 0 --load-checkpoint <path> --eval-only`.

## Baseline

- **DM best val:** 3.622% (T_max=36 MSE-only)
- **DM best TEST (full-eval):** 4.117% (gojo lr=4.8e-4 T_max=30, PR #3308)
- **Target: val < 3.622%**
- **Reproduce:** `cd target/icml2026 && SENPAI_TIMEOUT_MINUTES=600 python train.py --dataset drivaerml --optimizer adamw --lr 5e-4 --cosine-t-max 36 --grad-clip 0.5 --enable-fourier --model-layers 4 --model-hidden-dim 512 --model-heads 8 --epochs 999 --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200 --ema-decay 0.9995 --ema-mode fixed --no-compile-model --save-checkpoint`

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT franky:

## Results

### DrivAerML: gc=0.3 + lr=4.8e-4 + T_max=36 MSE-only

| Metric | Baseline (gc=0.5) | This run (gc=0.3) | Delta |
|--------|-------------------|-------------------|-------|
| **val_primary/surface_rel_l2_pct** | **3.622%** (ep 874) | **3.754%** (ep 858) | **+3.6% (worse)** |
| test_primary/surface_rel_l2_pct (batch-limited) | 4.631% | 4.444% | -4.0% |
| Best TEST (full-eval, ref) | 4.117% | N/A (no full-eval, val missed target) | — |

- **Target:** val < 3.622% — **NOT MET**
- **W&B run:** `p37o4bft` ([link](https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/p37o4bft))
- **Total epochs:** 863 (600-min timeout)
- **Best epoch:** 858

### Command used

```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 SENPAI_TIMEOUT_MINUTES=600 python train.py \
  --dataset drivaerml \
  --optimizer adamw \
  --lr 4.8e-4 \
  --cosine-t-max 36 \
  --grad-clip 0.3 \
  --enable-fourier \
  --model-layers 4 --model-hidden-dim 512 --model-heads 8 \
  --epochs 999 --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 --max-eval-batches 200 \
  --ema-decay 0.9995 --ema-mode fixed \
  --no-compile-model --save-checkpoint \
  --wandb-group dm-lr48e4-tmax36 \
  --agent franky \
  --wandb-name "franky/dm-lr48e4-tmax36-gc03"
```

### What happened

The softer gradient clipping (gc=0.3 vs gc=0.5) with lr=4.8e-4 **did not improve** over the baseline. Val best was 3.754% vs the baseline 3.622% — a +3.6% relative regression. The hypothesis that gc=0.3 would stabilize gradient spikes at higher LR and find a deeper basin did not materialize; instead, the tighter clipping appears to have over-constrained optimization.

Early training showed nearly all steps hitting the clip threshold (392-394/394), suggesting gc=0.3 is too aggressive for this LR — gradients are being clipped almost constantly in early epochs, likely slowing convergence. The grad clip events dropped over training as gradients naturally shrink, but the early-training constraint may have prevented the model from reaching the same basin.

Interestingly, the batch-limited test metric (4.444%) improved over baseline (4.631%), but this is likely noise from the batch-limited evaluation and the val metric is the reliable signal.

No full-eval run since val didn't beat the baseline target.

### Suggested follow-ups

- **gc=0.4** as a middle ground between 0.3 and 0.5 may be worth trying — gc=0.3 was too aggressive, but there could be a sweet spot.
- The near-constant clipping in early epochs suggests the real bottleneck is early optimization dynamics. A **warmup-aware clip schedule** (higher gc early, lower gc late) could help.
- The baseline gc=0.5 at lr=5e-4 still appears to be the better pairing. The lr=4.8e-4 compound may benefit more from other hyperparameter adjustments than clip strength.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #3396 because ADVISOR: Closing — gc=0.3 over-constrains early optimization, val=3.754% is +3.6% above baseline. The diagnostic (392-394/394 steps clipped in early epochs) confirms the threshold is too tight at this LR. The gc=0.5 default remains optimal for the lr=4.8e-4 compound. Thank you for the thorough analysis, franky..
